plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts

```diff
@@ -192,8 +209,159 @@ export const queryAiEndpoint = createMachine(
+	);
+	const fontPairing = FONT_PAIRINGS.find(
+		( pairing ) => pairing.title === fontPairingName
+	);
+
+	const globalStylesId = await resolveSelect(
+		coreStore
+		// @ts-ignore No types for this exist yet.
+	).__experimentalGetCurrentGlobalStylesId();
+
+	// @ts-ignore No types for this exist yet.
+	const { saveEntityRecord } = dispatch( coreStore );
+
+	await saveEntityRecord(
+		'root',
+		'globalStyles',
+		{
+			id: globalStylesId,
+			styles: mergeBaseAndUserConfigs(
+				colorPalette?.styles || {},
+				fontPairing?.styles || {}
+			),
+			settings: mergeBaseAndUserConfigs(
+				colorPalette?.settings || {},
+				fontPairing?.settings || {}
+			),
+		},
+		{
+			throwOnError: true,
+		}
+	);
+};
+
+// Update the current theme template
+const updateTemplate = async ( {
+	headerSlug,
+	businessSize,
+	homepageTemplateId,
+	footerSlug,
+}: {
+	headerSlug: string;
+	businessSize: 'SMB' | 'LB';
+	homepageTemplateId:
+		| keyof typeof SMALL_MEDIUM_BUSINESS_TEMPLATES
+		| keyof typeof LARGE_BUSINESS_TEMPLATES;
+	footerSlug: string;
+} ) => {
+	const patterns = ( await resolveSelect(
+		coreStore
+		// @ts-ignore No types for this exist yet.
+	).getBlockPatterns() ) as Pattern[];
+
+	const patternsByName = patternsToNameMap( patterns );
+
+	const headerPattern = patternsByName[ headerSlug ];
+	const footerPattern = patternsByName[ footerSlug ];
+
+	const homepageTemplate = getTemplatePatterns(
+		businessSize === 'SMB'
+			? SMALL_MEDIUM_BUSINESS_TEMPLATES[ homepageTemplateId ]
+			: LARGE_BUSINESS_TEMPLATES[ homepageTemplateId ],
+		patternsByName
+	);
+
+	const content = [ headerPattern, ...homepageTemplate, footerPattern ]
+		.filter( Boolean )
+		.map( ( pattern ) => pattern.content )
+		.join( '\n\n' );
+
+	const currentTemplate = await resolveSelect(
+		coreStore
+		// @ts-ignore No types for this exist yet.
+	).__experimentalGetTemplateForLink( '/' );
+
+	// @ts-ignore No types for this exist yet.
+	const { saveEntityRecord } = dispatch( coreStore );
+
+	await saveEntityRecord(
+		'postType',
+		currentTemplate.type,
+		{
+			id: currentTemplate.id,
+			content,
+		},
+		{
+			throwOnError: true,
+		}
+	);
+};
+
+export const assembleSite = async (
+	context: designWithAiStateMachineContext
+) => {
+	try {
+		await updateGlobalStyles( {
+			colorPaletteName: context.aiSuggestions.defaultColorPalette.name,
+			fontPairingName: context.aiSuggestions.fontPairing,
+		} );
+		recordEvent( 'customize_your_store_ai_update_global_styles_success' );
+	} catch ( error ) {
+		// TODO handle error
+		// eslint-disable-next-line no-console
+		console.error( error );
+		recordEvent(
+			'customize_your_store_ai_update_global_styles_response_error',
+			{
+				error: error instanceof Error ? error.message : 'unknown',
+			}
+		);
+	}
+
+	try {
+		await updateTemplate( {
+			headerSlug: context.aiSuggestions.header,
+			// TODO: Get from context
+			businessSize: 'SMB',
+			homepageTemplateId: 'template1',
```

Comment:
The business size and home template AI prompt still need to be done (https://github.com/woocommerce/woocommerce/issues/40162), so it is temporarily assumed to be SMB, and the first template is used.